mlflow/models/evaluation/default_evaluator.py

```diff
@@ -494,28 +499,51 @@ def _log_model_explainability(self):
             f: f2 for f, f2 in zip(self.feature_names, truncated_feature_names)
         }
 
-        sampled_X = shap.sample(self.X, sample_rows)
+        sampled_X = shap.sample(self.X, sample_rows, random_state=0)
+        if algorithm == "kernel":
+            background_X = shap.sample(self.X, sample_rows, random_state=3)
+        else:
+            background_X = None
 
         if isinstance(sampled_X, pd.DataFrame):
             # For some shap explainer, the plot will use the DataFrame column names instead of
             # using feature_names argument value. So rename the dataframe column names.
             sampled_X = sampled_X.rename(columns=truncated_feature_name_map, copy=False)
+            if background_X is not None:
+                background_X = background_X.rename(columns=truncated_feature_name_map, copy=False)
+        else:
+            # If sample_X is numpy array, convert to pandas dataframe.
+            sampled_X = pd.DataFrame(sampled_X, columns=truncated_feature_name_map)
+            if background_X is not None:
+                background_X = pd.DataFrame(background_X, columns=truncated_feature_name_map)
 
         if algorithm:
-            supported_algos = ["exact", "permutation", "partition"]
+            supported_algos = ["exact", "permutation", "partition", "kernel"]
             if algorithm not in supported_algos:
```

Comment:
We might want to do this validation right after `self.evaluator_config.get("explainability_algorithm", None)`.

Reply:
+1. supported_algos can be a module level constant like `_SUPPORTED_SHAP_ALGORITHMS = ("exact", "permutation", "partition", "kernel")`